servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -448,13 +449,22 @@ public DefaultSingleAddressHttpClientBuilder<U, R> appendConnectionFilter(
         requireNonNull(factory);
         connectionFilterFactory = appendConnectionFilter(connectionFilterFactory, factory);
         strategyComputation.add(factory);
+        ifHostHeaderHttpRequesterFilter(factory);
         return this;
     }
 
     @Override
     public DefaultSingleAddressHttpClientBuilder<U, R> appendConnectionFilter(
             final Predicate<StreamingHttpRequest> predicate, final StreamingHttpConnectionFilterFactory factory) {
```

Comment:
Is `HostHeaderHttpRequesterFilter` always guaranteed to be the outermost filter in the chain?

Reply:
There is no such requirement. The default variant that we add is always the last one, because we never know if users have another filter that manipulates with the `host` header or not. When users append it manually, they are free to place it at any position.

Reply:
OK, understood. It may be important to document that behavior so that users, who can now add the filter themselves, know to add it last.

Reply:
They can add it at any place as they may have other filters later in the chain that rely on the `host` header reading

Reply:
So there is no downside for us adding another instance of this filter? We have behavior associated with this filter, it seems like it should be described.

Reply:
From behavior point of view - no, you may have 2+ `HostHeaderHttpRequesterFilter` in the chain and result will be the same (first wins). Hence we don't try to prevent users from appending it multiple times.
For our internals, we tend to remove any unnecessary logic from runtime of request processing if we can at build time. So, I'm trying to avoid adding the default `HostHeaderHttpRequesterFilter` when there is no need for that.

Reply:
OK, sorry for the back and forth. When you say "first wins" do you mean outermost or innermost? If you mean outermost then there is there the possibility of the user added filter being ignored or not having the opportunity to have the effect they want? (calling back to the point that they may need to care about addition order).

Reply:
The first in a request processing chain:
```
client.request(req) -> HostHeaderHttpRequesterFilter(first.servicetalk.io) -> ... -> HostHeaderHttpRequesterFilter(last.servicetalk.io) -> LB -> connection filters -> netty
```
The serialized result will have `host: first.servicetalk.io`.

Reply:
This is clarified in javadoc:
> A filter which will set a HttpHeaderNames#HOST header with the fallback value if the header is not already present in HttpRequestMetaData.

Reply:
OK, the filter behaviour and order is understood. If the user adds their own `HostHeaderHttpRequesterFilter` instance how do they ensure that it will have effect and not be overridden by our adding another instance at the head of the chain if they did not place their instance at the head of the chain?

Reply:
We don't add to the head, our default `HostHeaderHttpRequesterFilter` always added last. This PR adds extra validation: if it sees that user appended `HostHeaderHttpRequesterFilter`, client builder won't add our default at all. Doesn't look like people should worry about anything in this case.